File: Javascript/342.4-的幂.js
/*
 * @lc app=leetcode.cn id=342 lang=javascript
 *
 * [342] 4的幂
 */

// @lc code=start
/**
 * @param {number} num
 * @return {boolean}
 */
var isPowerOfFour = function(num) {
     return /^1(00)*$/.test(num.toString(2));
     //test()方法用于检测字符串是否匹配某个模式，在这里模式为 /^1(00)*$/
};
// @lc code=end

File: Clanguage/342.4-的幂.c
/*
 * @lc app=leetcode.cn id=342 lang=c
 *
 * [342] 4的幂
 */

// @lc code=start
//判断任何次幂的c语言模板

bool isPowerOfFour(int num){
if(num==0){
    return false;
}

double m=log(num)/log(4),a,b,c;
c=m;
if(c-c>=0.5){
    c=c+1;
}
b=m-(int)c;
if(b<0){
    b=-b;
}
if(b<0.0000000001){
    return true;
}
return false;

}


// @lc code=end

